Refactor whole component, implement tests

StrongestNumber9 opened the pull request:
## Description
Refactor whole component, implement tests
<!-- Please include a summary of changes made in your pull request. -->

<!-- If you can't fill all check boxes in Checklists section, please explain why. -->

## Checklists

<!-- Fill check boxes before submitting the pull request. -->

### Testing

#### General

- [x] I have checked that my test files and functions have meaningful names.
- [x] I have checked that each test tests only a single behavior.
- [x] I have done happy tests.
- [x] I have tested only my own code.
- [ ] I have tested at least all public methods.

#### Assertions

- [x] I have checked that my tests use assertions and not runtime overhead.
- [x] I have checked that my tests end in assertions.
- [x] I have checked that there is no comparison statements in assertions.
- [x] I have checked that assertions are in tests and not in helper functions.
- [x] I have checked that assertions for iterables are outside of for loops and both sides of the iteration blocks.
- [x] I have checked that assertions are not tested inside consumers.

#### Testing Data

- [ ] I have tested algorithms and anything else with the possibility of unbound growth.
- [x] I have checked that all testing data is local and fully replaceable or reproducible or both.
- [x] I have checked that all test files are standalone.
- [ ] I have checked that all test-specific fake objects and classes are in the test directory.
- [x] I have checked that my tests do not contain anything related to customers, infrastructure or users.
- [x] I have checked that my tests do not contain non-generic information.
- [x] I have checked that my tests do not do external requests and are not privately or publicly routable.

#### Statements

- [x] I have checked that my tests do not use throws for exceptions.
- [x] I have checked that my tests do not use try-catch statements.
- [x] I have checked that my tests do not use if-else statements.

#### Java

- [x] I have checked that my tests for Java uses JUnit library.
- [x] I have checked that my tests for Java uses JUnit utilities for parameters.

#### Other

- [x] I have only tested public behavior and not private implementation details.
- [x] I have checked that my tests are not (partially) commented out.
- [x] I have checked that hand-crafted variables in assertions are used accordingly.
- [ ] I have tested [Object Equality](https://docs.oracle.com/javase/6/docs/api/java/lang/Object.html#equals%28java.lang.Object%29).
- [x] I have checked that I do not have any manual tests or I have a valid reason for them and I have explained it in the PR description.

### Code Quality

- [ ] I have checked that my code follows metrics set in Procedure: Class Metrics.
- [ ] I have checked that my code follows metrics set in Procedure: Method Metrics.
- [ ] I have checked that my code follows metrics set in Procedure: Object Quality.
- [x] I have checked that my code does not have any NULL values.
- [x] I have checked my code does not contain FIXME or TODO comments.

eemhu commented on src/main/java/com/teragrep/jla_06/RelpAppender.java:
```diff
@@ -32,267 +46,264 @@
-            syslog = syslog
-                    .withSDElement(event_id_48577)
-                    .withSDElement(origin_48577);
+    private static class SyslogRecordFactory {
```
would it be possible for `SyslogRecordFactory` to be in its own file?

StrongestNumber9 replied:
Implemented as its own class file in commit df4d0e7f39d7985e00429aa13cc773fbadf33fc9

eemhu commented on src/main/java/com/teragrep/jla_06/lib/syslog/SyslogRecordEventID.java:
```diff
@@ -0,0 +1,49 @@
+import com.teragrep.rlo_14.SyslogMessage;
+import java.util.UUID;
+
+public final class SyslogRecordEventID implements SyslogRecord {
```
perhaps SyslogRecord**With**EventId?

eemhu commented on src/main/java/com/teragrep/jla_06/lib/syslog/SyslogRecordPayload.java:
```diff
@@ -0,0 +1,37 @@
+
+import com.teragrep.rlo_14.SyslogMessage;
+
+public final class SyslogRecordPayload implements SyslogRecord {
```
same here about the naming

StrongestNumber9 commented on src/main/java/com/teragrep/jla_06/lib/syslog/SyslogRecordOrigin.java:
```diff
@@ -0,0 +1,40 @@
+import com.teragrep.rlo_14.SDElement;
+import com.teragrep.rlo_14.SyslogMessage;
+
+public final class SyslogRecordOrigin implements SyslogRecord {
```
I see the problem, refactored to "SyslogRecordWithX" version

eemhu commented on src/test/java/com/teragrep/jla_06/RelpLogAppenderTest.java:
```diff
@@ -0,0 +1,181 @@
+            String testIterationValue = matcher.group(1);
+
+            Assertions.assertDoesNotThrow(() -> {
+                Integer.parseInt(testIterationValue);
```
This could be written as
```
int testIteration = Assertions.assertDoesNotThrow(() -> Integer.parseInt(testIterationValue));
```

to avoid calling parseInt twice.

StrongestNumber9 commented on src/test/java/com/teragrep/jla_06/server/TestServerFactory.java:
```diff
@@ -0,0 +1,76 @@
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+public class TestServerFactory {
```
Fixed in commit 1302bceec1704a340755885159e104eb4a5e06fc

StrongestNumber9 commented on src/test/java/com/teragrep/jla_06/server/TestServer.java:
```diff
@@ -0,0 +1,50 @@
+
+import java.util.concurrent.ExecutorService;
+
+public class TestServer implements Runnable, AutoCloseable {
```
Fixed in commit 1302bceec1704a340755885159e104eb4a5e06fc

eemhu commented on README.adoc:
```diff
@@ -75,6 +81,30 @@ log4j2.shutdownHookEnabled=false
+|===
+
+|appender.RelpAppender.rebindEnabled
+|Whether rebinding the RELP connection is enablaed
```
typo -> enabled

kortemik commented:
> also, would equals and hashCode implementations be necessary for the new objects?

SyslogMessage included within SyslogRecord is mutable and

StrongestNumber9 commented:
> also, would equals and hashCode implementations be necessary for the new objects?

Implemented equals and hashCode for SyslogRecord*, added tests for them
